### PR TITLE
Feature/39/create user auctions service real impl

### PR DESCRIPTION
**Pull request by KaliszukTomasz**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
 node_modules/
 build/
 **.eslintrc
+build/contracts/Migrations.json
+build/contracts/Auction.json
```
Comment (lukaszkinder) [inline .gitignore]: this not needed

File: src/app/user/marketplaces-user/marketplaces-user.component.ts [hunk 2]
```diff
@@ -10,12 +10,17 @@ import { MarketplaceService } from '../service/marketplace-service';
 export class MarketplacesUserComponent implements OnInit {
 
   filteredMarketplaces: MarketplaceMin[] = [];
-
+  marketplaces: MarketplaceMin[] = [];
   constructor(
     @Inject('MarketplaceService') private readonly marketplaceService: MarketplaceService,
-  ) { this.getMarketplacesMatchedToUser(); }
+  ) { this.marketplaceService.getAllMarketplaces().subscribe(
+    res => this.filteredMarketplaces = res
```
Comment (lukaszkinder) [inline src/app/user/marketplaces-user/marketplaces-user.component.ts]: move to ngOnInit

File: src/app/user/service/auction-impl.service.ts [hunk 3]
```diff
@@ -1,23 +1,112 @@
...
+      .then((ContractAbstraction) => {
+        this.AuctionContract = ContractAbstraction;
+      });
+  }
 
   addOffer(offer: number, auctionAddress: string): Observable<boolean> {
-    return;
+    let isCompleteSubject = new Subject<boolean>();
+    isCompleteSubject.next(false);
+    var instance = this.AuctionContract.at(auctionAddress);
```
Comment (lukaszkinder) [inline src/app/user/service/auction-impl.service.ts]: what if address is invalid?

File: src/app/user/service/marketplace-impl.service.ts [hunk 4]
```diff
@@ -1,27 +1,374 @@
...
+                  )
+                });
+              });
+            }
+          }
+        });
+        return instance;
+      });
+
+    } catch (e) {
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: try catch

File: src/app/user/service/marketplace-impl.service.ts [hunk 5]
```diff
@@ -1,27 +1,374 @@
...
+
+  auctionToDetails: Auction = new Auction();
+
+  constructor(
+    private userService: UserService,
+    private web3Service: Web3Service,
+    @Inject('AuctionService') private readonly auctionService: AuctionService,
+  ) {
+
+    this.userService.getCurrentUser().subscribe(user => this.currentUser = user);
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: all this promisses should be chained - and if all promisses return success then call update method

File: src/app/user/service/marketplace-impl.service.ts [hunk 6]
```diff
@@ -1,27 +1,374 @@
...
+  //       console.log("ended winnerAddress: " + winnerAddress);
+  //       console.log("ended winningPrice: " + winningPrice);
+
+  //       auction.lastOffer = winningPrice;
+  //       auction.winner = winnerAddress;
+
+  //       // this.marketplacesStorage.find(MP => MP.address == auction.auctionAddress)
+  //       //   .auctions.find(auct => auct.auctionAddress == auction.auctionAddress).lastOffer = winningPrice;
+
+  //       // this.marketplacesStorage.find(MP => MP.address == auction.auctionAddress)
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: remove commented code

File: src/app/user/service/marketplace-impl.service.ts [hunk 7]
```diff
@@ -1,27 +1,374 @@
...
+              if (eventResult[i].blockNumber > mapWithLastRegister[userToEraseAddress]) {
+                var userIndex = marketplaceStorage.registeredUsers.findIndex(user => user.address == userToEraseAddress);
+                marketplaceStorage.registeredUsers.splice(userIndex, 1);
+              }
+            }
+          }
+        })
+      })
+      return 1;
+    } catch (e) {
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: try catch

File: src/app/user/service/marketplace-impl.service.ts [hunk 8]
```diff
@@ -1,27 +1,374 @@
...
+          auction.auctionName = this.web3Service.getWeb3Config().toAscii(eventResult[i].args.auctionName);
+          auction.lastOffer = 0;
+          auction.parentMarketPlace = eventResult[i].args.parentMarketPlace;
+          auction.owner = eventResult[i].args.createdBy;
+          auction.auctionCanceledFlag = false;
+          auction.auctionEndedFlag = false;
+          auction.winner = "no one was bidding yet";
+
+
+          var MPSIndex: number = this.marketplacesStorage.findIndex(MPS =>
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: what if not found?

File: src/app/user/service/marketplace-impl.service.ts [hunk 9]
```diff
@@ -1,27 +1,374 @@
...
+          // console.log("log 2" + auction);
+          // console.log("ilosc mp: " + marketplaceStorage.address + " " + i + this.marketplacesStorage[MPSIndex].auctions);
+
+
+
+        }
+      })
+
+    })
+    return 1;
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: why?

File: src/app/user/service/marketplace-impl.service.ts [hunk 10]
```diff
@@ -1,27 +1,374 @@
...
+
+  //     res(auction);
+
+  //   })
+
+  // }
+
+  // async updateAuctionCanceledEnded(auction: Auction) {
+
+  //   var instance = this.AuctionContract.at(auction.auctionAddress);
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: remove

File: src/app/user/service/marketplace-impl.service.ts [hunk 11]
```diff
@@ -1,27 +1,374 @@
...
+  //   //   }
+
+  //   // })
+  // }
+
+  async updateUsersInMarkets() {
+
+    var mapWithLastRegister: { [key: string]: number; } = {};
+
+    try {
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: remove try ..

File: src/app/user/service/auction-impl.service.ts (hunk 3 above)
Comment (KaliszukTomasz) [reply src/app/user/service/auction-impl.service.ts]: Trello task 45

File: src/app/user/service/marketplace-impl.service.ts (hunk 8 above)
Comment (KaliszukTomasz) [reply src/app/user/service/marketplace-impl.service.ts]: Trello task 45

File: .gitignore (hunk 1 above)
Comment (KaliszukTomasz) [reply .gitignore]: ok

File: src/app/user/marketplaces-user/marketplaces-user.component.ts (hunk 2 above)
Comment (KaliszukTomasz) [reply src/app/user/marketplaces-user/marketplaces-user.component.ts]: ok

File: src/app/user/service/marketplace-impl.service.ts (hunk 4 above)
Comment (KaliszukTomasz) [reply src/app/user/service/marketplace-impl.service.ts]: ok

File: src/app/user/service/marketplace-impl.service.ts (hunk 5 above)
Comment (KaliszukTomasz) [reply src/app/user/service/marketplace-impl.service.ts]: UpdateMarkets in constructor is not needed.

File: src/app/user/service/auction-impl.service.ts (hunk 3 above)
Comment (KaliszukTomasz) [reply src/app/user/service/auction-impl.service.ts]: Trello task 45

**Pull request merged**